### PR TITLE
T2D-116 Added variation accessioning code

### PR DESCRIPTION
**Pull request by jorizci**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/ampt2d/accession/ApplicationConstants.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession;
+
+public class ApplicationConstants {
+
+    public static final String APPLICATION_INSTANCE_ID = "accessioning.instance.id";
```
Comment (pabarcgar) [inline src/main/java/uk/ac/ebi/ampt2d/accession/ApplicationConstants.java]: I commented this in other project, I know that @jorizci sometimes likes to have the constants grouped in a class. I see them belonging to other classes like EvaAccessionApplication. Any other opinion? I din't dislike it so much, so is OK if this class remains.

Comment (pabarcgar) [reply src/main/java/uk/ac/ebi/ampt2d/accession/ApplicationConstants.java]: Actually, they are being used in VariantAccessioningConfiguration, so maybe that is the place

File: src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    @Autowired
+    private VariantAccessioningRepository accessioningObjectRepository;
+
+    @Test
+    public void testRestApi() {
+        String url = "/v1/variant";
+        HttpEntity<Object> requestEntity = new HttpEntity<>(getListOfVariantMessages());
+        ResponseEntity<Map> response = testRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Map.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(2, response.getBody().size());
```
Comment (pabarcgar) [inline src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java]: The response in the tests in this class should be checked in more deep

File: src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    private TestRestTemplate testRestTemplate;
+
+    @Autowired
+    private VariantAccessioningRepository accessioningObjectRepository;
+
+    @Test
+    public void testRestApi() {
+        String url = "/v1/variant";
+        HttpEntity<Object> requestEntity = new HttpEntity<>(getListOfVariantMessages());
+        ResponseEntity<Map> response = testRestTemplate.exchange(url, HttpMethod.POST, requestEntity, Map.class);
```
[the remark this answers is not in the document]
Comment (jorizci) [reply src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java]: No, if it has a structure it should be defined.

File: src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java (hunk 2 above)
Comment (jorizci) [reply src/test/java/uk/ac/ebi/ampt2d/accession/variant/VariantAccessioningRestControllerTest.java]: The Json conversion is handled with Spring converters so no logic to check there. The important part is that it returns two entities.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+accessioning.namespace=AMP
```
[the remark this answers is not in the document]
Comment (jorizci) [reply src/main/resources/application.properties]: SS

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+# org.springframework.transaction -> trans
```
[the remark this answers is not in the document]
Comment (jorizci) [reply src/main/resources/application.properties]: Yup

**Pull request merged**